ls1intum/Artemis
# Add paging to plagiarism results

### Checklist

### Motivation and Context
For exercises with a large amount of submissions, displaying all detected plagiarism matches is quite expensive and slows down the client application. This is why this PR adds paging functionality to show a max. number of 100 matches at once.

### Description
I added a small paging panel at the bottom of the sidebar to navigate through multiple pages. Each page will show 100 results.

### Steps for Testing

**Note**: For testing purposes, I decreased the paging size to 10.

The following should work for **modeling**, **text**, and **programming** exercises. Make sure you have multiple similar submissions to generate enough results.

1. Log in to Artemis
2. Navigate to Course Administration
3. Select a course
4. Go to Exercises
5. Navigate to the details page of an exercise by clicking 'View'
6. Click 'Check Plagiarism'
7. Click 'Check Plagiarism' on the plagiarism page and wait for the results
8. If you have more than 10 results, they should be paged correctly and only 10 results should be displayed at once

### Test Coverage
* TODO: plagiarism-sidebar.component.spec.ts 25%

### Screenshots
<img width="1680" alt="Screenshot 2020-12-22 at 22 36 46" src="https://user-images.githubusercontent.com/21973908/102935426-684bf380-44a6-11eb-9a0e-1c0e5bf00c0a.png">
